.github/workflows/go-test.yml

```diff
@@ -0,0 +1,25 @@
+name: go-test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request: { }
+
+# Lifted from https://github.com/airbnb/prometheus/blob/main/.circleci/config.yml
+jobs:
+  test:
+    name: go test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: prometheus/setup-environment
+        run: make promu
+      - name: run make
+        run: make GO_ONLY=1 test
+        env:
+          GOGC: 20
+          GOOPTS: "-p 2"
+          GOMAXPROCS: "2"
+          GO111MODULE: "on"
```

Comment:
Curious, are we sure this is running all the tests we want to run?

I don't see the `testdata/functions.test` file in the output (not sure if I should?). Just wanted to ask the question though

Reply:
I introduced a failure and the `make GO_ONLY=1 test` did fail with that error output.